tee-worker/litentry/core/vc-issuance/lc-vc-task-receiver/src/lib.rs

```diff
@@ -137,86 +140,102 @@ where
 	if let TrustedCall::request_vc(signer, who, assertion, maybe_key, _hash) =
 		trusted_call.call.clone()
 	{
-		let (mut state, _) = context.state_handler.load_cloned(&shard).map_err(|e| {
-			format!("Received error while trying to obtain sidechain state: {:?}", e)
-		})?;
-
-		state.execute_with(|| {
-			let key = match maybe_key {
-				Some(s) => s,
-				None => return Err("User shielding key not provided".to_string()),
-			};
-
-			let id_graph = IMT::get_id_graph(&who);
-			let assertion_networks = assertion.clone().get_supported_web3networks();
-			let identities: Vec<IdentityNetworkTuple> = id_graph
-				.into_iter()
-				.filter(|item| item.1.is_active())
-				.map(|item| {
-					let mut networks = item.1.web3networks.to_vec();
-					networks.retain(|n| assertion_networks.contains(n));
-					(item.0, networks)
-				})
-				.collect();
-
-			let signer = match signer.to_account_id() {
+		let key = match maybe_key {
+			Some(s) => s,
+			None => return Err("User shielding key not provided".to_string()),
+		};
+		let identities: Vec<IdentityNetworkTuple> = context
+			.state_handler
+			.execute_on_current(&shard, |state, _| {
+				let prefix_key = storage_map_key(
+					"IdentityManagement",
+					"IDGraphs",
+					&who,
+					&StorageHasher::Blake2_128Concat,
+				);
+				let mut id_graph =
+					state.iter_prefix::<Identity, IdentityContext<Runtime>>(&prefix_key).unwrap();
+				id_graph.sort_by(|a, b| {
+					let order = Ord::cmp(&a.1.link_block, &b.1.link_block);
+					if order == Ordering::Equal {
+						// Compare identities by their did formated string
+						Ord::cmp(&a.0.to_did().ok(), &b.0.to_did().ok())
+					} else {
+						order
+					}
+				});
+				let assertion_networks = assertion.clone().get_supported_web3networks();
+				let identities: Vec<IdentityNetworkTuple> = id_graph
+					.into_iter()
+					.filter(|item| item.1.is_active())
+					.map(|item| {
+						let mut networks = item.1.web3networks.to_vec();
+						networks.retain(|n| assertion_networks.contains(n));
+						(item.0, networks)
+					})
+					.collect();
+
+				identities
+			})
+			.map_err(|e| format!("Failed to fetch sidechain data due to: {:?}", e))?;
+
+		let signer = match signer.to_account_id() {
+			Some(s) => s,
+			None => return Err("Invalid signer account, failed to convert".to_string()),
+		};
```

Comment:
Is it possible to use `.ok_or` here for better readability? Other places too

Reply:
I initially tried using `unwrap_or` but I had some problems, So decided to use `match`, I can try it again.

Reply:
`.ok_or` works like a charm, updated ✅